src/app/providers.tsx

```diff
@@ -18,15 +19,17 @@ function Providers({ children }: React.PropsWithChildren) {
     <ThemeProvider defaultTheme="dark" attribute="data-theme">
       <QueryClientProvider client={client}>
         <ErrorProvider>
-          <WalletProvider>
-            <AppState>
-              <StakingStatsProvider>
-                <ReactQueryStreamedHydration>
-                  {children}
-                </ReactQueryStreamedHydration>
-              </StakingStatsProvider>
-            </AppState>
-          </WalletProvider>
+          <TomoProvider>
+            <WalletProvider>
```

Comment:
Theoretically, but in Tomo provider we manage the wallet lists, cosmos chains, wallet provider theme and all of that stuff. Related to the overall modal, connection, etc
Wallet provider is actually the provider of our wallets. It might be split to 2 different parts - BTC wallet and Cosmos wallet (as per @totraev request), though these instances live in the same wallet connection flow.

Reply:
@jrwbabylonlab Yes, it's necessary. In TomoProvider we configure Tomo itself, however WalletProvider (needs to be splitted into BTCWalletProvider and BBNWalletProvider) is our abstraction on top of Tomo's existing BTC wallet provider